Sources/FusionBluetooth_Common/BluetoothManager.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+/**
+ *  @struct Peripheral
+ *
+ *  @discussion The device data for peripheral
+ *
+ */
+public struct Peripheral: Equatable {
+
+    /**
+     *  @property name
+     *
+     *  @discussion The device name.
+     *
+     */
+    public let name: String?
+    
+    /**
+     *  @property uuid
+     *
+     *  @discussion The identifier in iOS and MAC address in Android.
+     *
+     */
+    public let uuid: String
+    
+     /**
+     *  @property isConnected
+     *
+     *  @discussion The state whether the peripheral is connected or not
+     *
+     */
+    public var isConnected: Bool
+    
+    public init(name: String?, uuid: String, isConnected: Bool) {
+        self.name = name
+        self.uuid = uuid
+        self.isConnected = isConnected
+    }
+}
+
+/**
+ *  @protocol BluetoothManagerProtocol
+ *
+ *  @discussion The delegate of managing Bluetooth.
+ *
+ */
+public protocol BluetoothManagerProtocol {
+
+	/*
+     * @method requestAuthorization:
+     *
+     * @discussion Requests the ACCESS_FINE_LOCATION permission. For only Android
+     */
+    func requestAuthorization()
+    
+    /*
+     * @method isScanning:
+     *
+     * @discussion Whether or not the central is currently scanning.
+     */
+	func isScanning() -> Bool
+	
+	/*
+     * @method checkState:
+     *
+     * @param receiver Returns whether the central device is enabled
+     *
+     * @discussion Reads the state of central device
+     */     
+	func checkState(receiver: @escaping (Bool) -> Void)
```

Comment:
Also naming, checkState asynchronously receiving Bool is not clear enough, if it's checking whether bluetooth is enabled then it should be called like: checkAuthorization or isAuthorized.

Also, what is not clear for me is why do I need an explicit "check", I also saw this in the Location API. Wouldn't be enough just to "requestAuthorization" asynchronously, wait for an answer and call routines if it's enabled. Additionally, if I want to "startDiscovery", may be it would be possible just to pass a lambda so that authorization can be done in the background. Could you please share your thought here. Thanks in advance

Reply:
yeah I renamed

Reply:
1. I renamed the function to isEnabled()
2. it was not to check the authorization status.
3. It is not normal sense to wait for the authorization enabled in block.
4. We can only check the one-time via isAuthorized.
5. I added the isAuthorized()